### ISSUE TITLE
Bettercap API stalls and causes UI to not function

### ISSUE THREAD

**Issue by nicholaschiang** [bettercap/bettercap#811]
*Description of the bug or feature request*

I just installed the latest Bettercap version on my Pop_OS! 20.10 system (essentially Ubuntu 20.10 with a few tweaks from System76) and haven't been able to get the local `http-ui` to work properly.

After following [the installation instructions](https://www.bettercap.org/installation/#compiling-from-sources), I ran:

```
$ sudo bettercap -eval "caplets.update; ui.update; q"
$ sudo bettercap -caplet http-ui
```

I then opened up the default http://127.0.0.1:80, entered the default login credentials (`user` and `pass`), and then got stuck on that login page (and the interactive terminal froze as well; I had to kill the process manually).

This failed for me no matter what browser I used (Firefox 83.0 and Chrome 87.0.4280.66).

### Environment

Please provide:

* Bettercap version you are using ( `bettercap -version` ): `bettercap v2.28 (built for linux amd64 with go1.14.7)`
* OS version and architecture you are using: Pop_OS! 20.10 64-bit (essentially Ubuntu 20.10)
* Go version if building from sources: `go version go1.14.7 linux/amd64`
* Command line arguments you are using: `-caplet http-ui`
* Caplet code you are using or the interactive session commands: N/A
* **Full debug output** while reproducing the issue ( `bettercap -debug ...` ):

```
$ sudo bettercap -debug -caplet http-ui
bettercap v2.28 (built for linux amd64 with go1.14.7) [type 'help' for a list of commands]

192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [dbg] FindGateway(wlp3s0) [cmd=ip opts=[route] parser=^(default|[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)\svia\s([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)\sdev\s(\S+).*$]
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [dbg] FindGateway(wlp3s0) output:
default via 192.168.10.1 dev wlp3s0 proto dhcp metric 600 
169.254.0.0/16 dev wlp3s0 scope link metric 1000 
192.168.10.0/24 dev wlp3s0 proto kernel scope link src 192.168.10.179 metric 600
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [dbg] FindGateway(wlp3s0) line 'default via 192.168.10.1 dev wlp3s0 proto dhcp metric 600' matched with [default via 192.168.10.1 dev wlp3s0 proto dhcp metric 600 default 192.168.10.1 wlp3s0]
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [dbg] gateway is 192.168.10.1[20:76:93:55:44:1f]
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [session.started] {session.started 2021-01-12 18:54:09.406735258 -0800 PST m=+0.102303346 <nil>}
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [mod.started] events.stream
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [mod.started] api.rest
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [mod.started] http.server
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [inf] http.server starting on http://127.0.0.1:80
192.168.10.0/24 > 192.168.10.179  » [18:54:09] [sys.log] [inf] api.rest api server starting on http://127.0.0.1:8081
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/styles.92a454e3ed7c5bffbed3.css
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/runtime.26209474bfa8dc87a77c.js
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/scripts.68228f7acaf48355d8cf.js
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/main.ec83a59338123ba3bb5d.js
192.168.10.0/24 > 192.168.10.179  » [18:54:19] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/polyfills.681b44b3f292a3819611.js
192.168.10.0/24 > 192.168.10.179  » [18:54:20] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/assets/images/logo.png
192.168.10.0/24 > 192.168.10.179  » [18:54:20] [sys.log] [dbg] http.server 127.0.0.1 GET 127.0.0.1/assets/images/logo.png
192.168.10.0/24 > 192.168.10.179  »  
```

### Steps to Reproduce

1. Follow [these installation instructions](https://www.bettercap.org/installation/#compiling-from-sources) on Pop_OS! 20.10 (or just Ubuntu 20.10 for that matter).
2. Run `sudo bettercap -eval "caplets.update; ui.update; q"` as [recommended here](https://www.bettercap.org/usage/#web-ui).
3. Run `sudo bettercap -caplet http-ui`.
4. Visit http://127.0.0.1:80 and type in the default login info (`user` and `pass`).
5. Notice how it stalls and nothing changes.

**Expected behavior:** *What you expected to happen*

The API should respond (it didn't respond to any requests, even when made with cURL) and the `http-ui` should redirect to the events page once logged in.

**Actual behavior:** *What actually happened*

See above issue description.

**Comment by nicholaschiang** [comment]
I just checked out to the `v2.28` tag and recompiled. It seems to be working fine now.

It seems as if this is just broken on `master` (which I thought would be relatively stable and tested... guess not). You should rollback those changes until this issue is addressed IMHO.

**Comment by curious-ninja** [comment]
I just built bettercap with the following commit:
https://github.com/bettercap/bettercap/commit/ac4b1f6e9e4faa54533dbc5b1cb68d3fbe3b3bc3

And that's where it seems the api login screen broke.

The last commit where the login works is:
https://github.com/bettercap/bettercap/commit/8acf81f61d916c0d8ab71c1c65ab65d3a4a13749

There was mention [here](https://github.com/bettercap/bettercap/issues/803#issuecomment-757003304) of the web-ui being broken after a set of unrelated (not committed) code. But may have been some confusion there.

**Comment by curious-ninja** [comment]
Possibly caused by lines 70 and 71 of network/wifi.go:
https://github.com/bettercap/bettercap/commit/ac4b1f6e9e4faa54533dbc5b1cb68d3fbe3b3bc3

**Comment by bonedaddy** [comment]
> Possibly caused by lines 70 and 71 of network/wifi.go:
> [ac4b1f6](https://github.com/bettercap/bettercap/commit/ac4b1f6e9e4faa54533dbc5b1cb68d3fbe3b3bc3)

This does appear to be causing the issue, I tried building without this change and it unfroze the webui.

**Comment by evilsocket** [comment]
fixed by https://github.com/bettercap/bettercap/pull/812 ... thanks @bonedaddy

**Comment by bonedaddy** [comment]
> fixed by #812 ... thanks @bonedaddy

No problem, apologies for any issues caused.
